src/AlgebraicGeometry/Schemes/AffineVariety/Objects/Constructors.jl

```diff
@@ -0,0 +1,64 @@
+export affine_variety
+
+########################################################
+# (1) Generic constructors
+########################################################
+@doc Markdown.doc"""
+    affine_variety(X::Spec; check::Bool=true) -> AffineVariety
+
+Convert ``X`` to an affine variety.
+
+If check is set, then compute the reduced scheme of `X` first.
+"""
+function affine_variety(X::Spec{<:Field}; check::Bool=true)
+  check  ||  AffineVariety(X, check=check)
+  Xred,_ = reduced_scheme(X)
+  return AffineVariety(Xred, check=check)
+end
+
+@doc Markdown.doc"""
+    affine_variety(I::MPolyIdeal; check=true) -> AffineVariety
+
+Return the affine variety defined by the prime ideal ``I``.
+
+Since our varieties are irreducible, we check that ``I`` stays prime when
+viewed over the algebraic closure. This is an expensive check that can be disabled.
+"""
+affine_variety(I::MPolyIdeal; check=true) = AffineVariety(Spec(quo(base_ring(I),I)[1]), check=check)
+
+@doc Markdown.doc"""
+    affine_variety(R::Ring; check=true)
+
+Return the affine variety defined by ``R``.
+
+We require that ``R`` is a finitely generated algebra over a field ``k`` and
+moreover that the base change of ``R`` to the algebraic closure ``\bar k``
+is an integral domain.
+"""
+affine_variety(R::MPolyAnyRing; check=true) = AffineVariety(Spec(R), check=check)
+
+
+function affine_variety(f::MPolyRingElem; check=true)
+  if check
+    is_irreducible(f) || error("not a variety")
```

Comment:
As input is a polynomial, it would be better to be more explicit: "polynomial is reducible"
It would make it easier for the user to track down the point of the error message.